### DISCUSSION TITLE
Syncing only shared folders

### DISCUSSION THREAD

**Discussion by aothmane-control** [abraunegg/onedrive#2711] [Q&A] [answered]
I would like to sync only folders that other people from my organization have shared with me. I have put into the file business_shared_folders the shared folders I want to sync. Unfortunately I have two problems:

1.  running the command `onedrive --list-shared-folders --confdir=my_conf_path` I get an uncomplete list of folders. Some shared folders (all of them are from people from my organization) are simply not shown.
2.  How to exclude in the synchronization all OneDrive content except the shared folders? If I use sync_list or skip_dir then a resync takes for ever.

**Comment by abraunegg** [comment, marked as answer]
@aothmane-control 
> I would like to sync only folders that other people from my organization have shared with me

Firstly I am going to preface this response with that using *any* v2.4.x application version to achieve your desired outcome is not going to work for you. 

Given where the development cycle is at, as a fist step, please *uninstall* the v2.4.x version you are currently using, and install v2.5.0-rc2. To do this, please read: https://github.com/abraunegg/onedrive/discussions/2710

As such, all of my answers below will refer to v2.5.0-rc2 information - sorry.

OneDrive Business Shared Folder handling has been 100% re-written in v2.5.x which should support much better your desired configuration.

> running the command onedrive --list-shared-folders --confdir=my_conf_path I get an uncomplete list of folders. Some shared folders (all of them are from people from my organization) are simply not shown.

When you perform this function using v2.5.0 you should get a complete list as this function was 100% re-written:
```
--list-shared-items
      List OneDrive Business Shared Items
```
**example:**
```
Listing available OneDrive Business Shared Items:

-----------------------------------------------------------------------------------
Shared File:     large_document_shared.docx
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared File:     no_download_access.docx
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared File:     online_access_only.txt
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared File:     read_only.txt
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared File:     qewrqwerwqer.txt
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared File:     dummy_file_to_share.docx
Shared By:       testuser2 testuser2 (testuser2@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   Sub Folder 2
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared File:     file to share.docx
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   Top Folder
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   my_shared_folder
Shared By:       testuser2 testuser2 (testuser2@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   Jenkins
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   Jenkins
Shared By:       testuser2 testuser2 (testuser2@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   Empty_Folder
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   100K_Files
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   Foo (Core)
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
Shared Folder:   my-library-folder
Shared By:       test user (testuser@my.org.tld)
-----------------------------------------------------------------------------------
```
> How to exclude in the synchronization all OneDrive content except the shared folders? If I use sync_list or skip_dir ....

Once you add the desired folders to your account, as per the updated documentation for this feature, you simply need to add the shared folders you wish to sync to your 'sync_list' file, as everything else will be excluded.

You also *may* not need to do this, if you are using a OneDrive Account that has *zero* other files to begin with - when you add the Shared Folder Link, there is only those links to sync ..... something to potentially think about.

> .... than a resync takes for ever.

The v2.5.x client has been 100% totally re-written with many performance improvements including this.

### Actions for you
1. Upgrade your client to v2.5.0-rc2
2. Read *all* v2.5.0-rc2 documentation
3. Reconfigure your online state to be compatible with v2.5.0 usage.
4. Test, test and re-test using `--dry-run` to ensure you are happy with what is occurring to achieve your desired state, then remove `--dry-run`

### References
* https://github.com/abraunegg/onedrive/pull/2709
* v2.5.0-rc2 Documentation: https://github.com/abraunegg/onedrive/tree/onedrive-v2.5.0-release-candidate-2/docs
* How to sync OneDrive Business Shared Items: https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-2/docs/business-shared-items.md

**Reply by aothmane-control** [reply]
I have made the following observation: Consider the following situation: Person A shares with a me a folder `tmp`. The folders is the onedrive folder of Person A under `/test/tmp`. If I add the shortcut for the shared folder to my folders I can see it in the browser. If I want to sync the folder I tried to put `/tmp/` in the sync_list file. However, this did not work. The folder was not synced. I had to put `/test/tmp/` in order to be able to download the folder. Is this an issue with the onedrive client or a limitation from microsoft?

**Reply by abraunegg** [reply]
Please raise a bug against v2.5.0-rc2 as this is not the expected outcome.
